### PR TITLE
Update for recent platform features

### PR DESCRIPTION
**Pull request by zyro**

### CODE REVIEW

File: sdk/src/main/java/io/gameup/android/GameUpSession.java [hunk 1]
```diff
@@ -611,4 +810,168 @@ private PurchaseVerification purchaseVerify(
...
+        return RequestSender.send(request, MessageList.class);
+    }
+
+    /**
+     * Mark a message as read, and return its metadata, including its body.
+     *
+     * @param messageId The identifier of the message to mark as read.
+     * @return A corresponding Message instance.
+     */
+    public Message messageRead(final @NonNull String messageId) {
```
Comment (mofirouz) [inline sdk/src/main/java/io/gameup/android/GameUpSession.java]: Shouldn't this be MessageGet - Reading a message is just the side effect and not the main intention of the endpoint correct?

File: sdk/src/main/java/io/gameup/android/GameUpSession.java [hunk 2]
```diff
@@ -611,4 +810,168 @@ private PurchaseVerification purchaseVerify(
...
+    /**
+     * Mark a message as read, and return its metadata, optionally including its
+     * body.
+     *
+     * @param messageId The identifier of the message to mark as read.
+     * @param withBody true if the message body should be included,
+     *                 false otherwise.
+     * @return A corresponding Message instance.
+     */
+    public Message messageRead(final @NonNull String messageId,
```
Comment (mofirouz) [inline sdk/src/main/java/io/gameup/android/GameUpSession.java]: Same as above

File: sdk/src/main/java/io/gameup/android/json/GamerUpdate.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+/**
+ * Represents a request to update the account or profile data of a gamer.
+ */
+@Data
+@AllArgsConstructor(suppressConstructorProperties = true)
+public class GamerUpdate {
```
Comment (mofirouz) [inline sdk/src/main/java/io/gameup/android/json/GamerUpdate.java]: Why is this class needed - can we not just use the nickname as method params?

**Comment by mofirouz** [comment]
I've reviewed this - everything looks really good, just some minor questions :)

File: sdk/src/main/java/io/gameup/android/GameUpSession.java (hunk 1 above)
Comment (zyro) [inline sdk/src/main/java/io/gameup/android/GameUpSession.java]: Actually, the `withBody` flag toggles this between `read` and `read & get`, so 'read' is really the constant here. Does that make sense?

File: sdk/src/main/java/io/gameup/android/GameUpSession.java (hunk 2 above)
Comment (zyro) [inline sdk/src/main/java/io/gameup/android/GameUpSession.java]: Discussion above.

File: sdk/src/main/java/io/gameup/android/json/GamerUpdate.java (hunk 3 above)
Comment (zyro) [inline sdk/src/main/java/io/gameup/android/json/GamerUpdate.java]: We do, but then we construct the update object from it. I don't want to handle manually constructing and validating a JSON string.

**Comment by mofirouz** [comment]
Okies cool. Let's merge this. :+1:

**Pull request closed without merging**